### ISSUE TITLE
Performance issue in the definition of level_select, fsaf_layers.py(P1)

### ISSUE THREAD

**Issue by DLPerf** [xuannianz/FSAF#39]
Hello, I found a performance issue in the definition of level_select, fsaf_layers.py, [`tf.reduce_prod(feature_shapes, axis=-1)`](https://github.com/xuannianz/FSAF/blob/791ca41187eec25918b2626322c43c9ab7c06f61/fsaf_layers.py#L35) will be created repeatedly during program execution, resulting in reduced efficiency. I think event_hidden_feats_tile and event_hidden_feats_reshape should be created before the loop.

The same issue exists in:

- [`tf.reduce_prod` in line 36](https://github.com/xuannianz/FSAF/blob/791ca41187eec25918b2626322c43c9ab7c06f61/yolo/fsaf_layers.py#L36)

Looking forward to your reply. Btw, I am very glad to create a PR to fix it if you are too busy.
